marine.c

```diff
@@ -579,24 +579,49 @@ WS_DLL_PUBLIC marine_result *marine_dissect_packet(int filter_id, unsigned char
     return result;
 }
 
+int compile_bpf(char *bpf, struct bpf_program *fcode, char *err_msg) {
```

Comment:
I don't think there's a point in passing `err_msg` here since the error is predetermined (hardcoded in the function) and doesn't add any real value to the failure code.

This leaves us with two options:
1. Remove `err_msg` (I prefer this one)
2. Use [`pcap_geterr`](https://linux.die.net/man/3/pcap_compile) although I don't think it's needed.